File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IFaveLinksView.java
package dev.ragnarok.fenrir.mvp.view;

import java.util.List;

import dev.ragnarok.fenrir.model.FaveLink;
import dev.ragnarok.fenrir.mvp.core.IMvpView;
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView;

public interface IFaveLinksView extends IAccountDependencyView, IMvpView, IErrorView {
    void displayLinks(List<FaveLink> links);

    void notifyDataSetChanged();

    void notifyDataAdded(int position, int count);

    void displayRefreshing(boolean refreshing);

    void openLink(int accountId, FaveLink link);

    void notifyItemRemoved(int index);
}

File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IFaveLinksView.kt
package dev.ragnarok.fenrir.mvp.view

import dev.ragnarok.fenrir.model.FaveLink
import dev.ragnarok.fenrir.mvp.core.IMvpView
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView

interface IFaveLinksView : IAccountDependencyView, IMvpView, IErrorView {
    fun displayLinks(links: List<FaveLink>)
    fun notifyDataSetChanged()
    fun notifyDataAdded(position: Int, count: Int)
    fun displayRefreshing(refreshing: Boolean)
    fun openLink(accountId: Int, link: FaveLink)
    fun notifyItemRemoved(index: Int)
}